src/dataflow/src/sink/kafka.rs

```diff
@@ -562,9 +596,8 @@ where
     };
 
     // We want exactly one worker to send all the data to the sink topic.
-    // This should already have been handled upstream (in render/mod.rs),
-    // so we can use `Pipeline` here.
-    let mut input = builder.new_input(stream, Pipeline);
+    let mut input = builder.new_input(&stream, Exchange::new(move |_| id.hashed()));
```

Comment:
I believe this will re-evaluate `id.hashed()` for each record. Probably best to compute once outside the closure and then jam the result in here.

Reply:
👍